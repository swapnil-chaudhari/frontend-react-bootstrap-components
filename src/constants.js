export const FILTER_PERIODS = {
    YEAR: 365,
    MONTH: 27,
    WEEK: 6,
    DAYS_90: 90,
};

export const SEARCH_TYPES = {
    ARTISTS: 'artists',
    TRACKS: 'tracks'
};

export const KEYS = {
    ENTER: 13,
};
