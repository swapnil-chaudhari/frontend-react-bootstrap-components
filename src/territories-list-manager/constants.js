export const TERRITORIES_KEYS = {
    ID: 'orchId',
    NAME: 'territoryName',
    CONTINENT: 'continent'
};
