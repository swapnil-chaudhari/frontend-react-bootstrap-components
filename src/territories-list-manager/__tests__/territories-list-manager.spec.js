import React from 'react';
import renderShallow from 'render-shallow';
import { oneLine } from 'common-tags';
import { findWithClass } from 'react-shallow-testutils';
import TerritoriesListManager from '../territories-list-manager';
import noop from 'src/utils/noop';
import { TERRITORIES_KEYS } from '../constants';

describe('<TerritoriesListManager>', () => {
    const classes = (element) => element.props.className.split(' ');

    const commonProps = {
        thisListType: 'foo',
        theOtherListType: 'bar',
        forward: true,
        header: 'Cleared For Sale',
        totalNumberOfCountries: 0,
        onUndoNotificationClose: noop,
        undoNotificationVisible: false,
        onTerritorySelect: noop,
        countries: [],
        selectedCountries: [],
        movedCountriesNumber: 0,
        onTerritoryMovedUndo: noop,
        onTerritorySearchClose: noop,
        onTerritoryMove: noop,
        onCountrySelect: noop,
        theOtherListLabel: 'the other list',
        formatMessage: msg => msg,
        messages: {},
        territoryIdKey: TERRITORIES_KEYS.ID,
        territoryNameKey: TERRITORIES_KEYS.NAME,
        continentNameKey: TERRITORIES_KEYS.CONTINENT
    };

    describe('when it renders', () => {
        const props = { ...commonProps };
        let component;

        beforeAll(() => {
            component = renderShallow(<TerritoriesListManager { ...props } />).output;
        });

        test('renders the correct className', () => {
            expect(classes(component)).toContain('TerritoriesListManager');
            expect(classes(component)).toContain('col-xs-6');
        });

        test('should contain 3 components', () => {
            expect(findWithClass(
                component,
                'TerritoriesListManager'
            ).props.children).toHaveLength(3);
        });

        test('should contain TerritoriesListManager-list', () => {
            expect(findWithClass(
                component,
                'TerritoriesListManager-list Opm-thin-grey-border Opm-border-radius'
            ).type).toEqual('div');
        });

        test(
            'should contain ListHeader component in TerritoriesListManager-list',
            () => {
                expect(findWithClass(
                    component,
                    'TerritoriesListManager-list Opm-thin-grey-border Opm-border-radius'
                )).toMatchSnapshot();
            }
        );

        test(
            'should hide TerritoriesListManager-undo-message when no changes available ',
            () => {
                expect(findWithClass(
                    component,
                    'TerritoriesListManager-undo-message Opm-thin-grey-border hide'
                )).toMatchSnapshot();
            }
        );

        test('should contain TerritoriesListManager-move-button ', () => {
            expect(component).toMatchSnapshot();
        });

        test('should hide MoveCountryButton when no selected countries', () => {
            const moveButton = findWithClass(
                component,
                'TerritoriesListManager-move-button Opm-border-radius'
            );
            expect(moveButton.props.className).toContain('hide');
        });
    });

    describe('when it renders with selected countries', () => {
        const props = {
            ...commonProps,
            selectedCountries: [{}, {}],
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<TerritoriesListManager { ...props } />).output;
        });

        test('should show MoveCountryButton when there are selected countries', () => {
            const moveButton = findWithClass(
                component,
                'TerritoriesListManager-move-button Opm-border-radius'
            );
            expect(moveButton.props.className).toContain('TerritoriesListManager-move-button');
            expect(moveButton.props.className).toContain('Opm-border-radius');
        });
    });

    describe(oneLine`when undoNotificationVisible is true
    kand movedCountriesNumber is greater than 0`, () => {
        let component;
        const props = {
            ...commonProps,
            undoNotificationVisible: true,
            movedCountriesNumber: 2
        };

        beforeAll(() => {
            component = renderShallow(<TerritoriesListManager { ...props } />).output;
        });

        test('should show TerritoriesListManager-undo-message', () => {
            expect(findWithClass(
                component,
                'TerritoriesListManager-undo-message Opm-thin-grey-border'
            )).toMatchSnapshot();
        });
    });

    describe('when search is triggered', () => {
        const props = {
            ...commonProps,
            territoryNameKey: 'countryName',
            totalNumberOfCountries: 1,
            countries: [{ countryName: 'Japan' }],
        };
        let component;
        let instance;
        let rerender;

        beforeAll(() => {
            ({ output: component, instance, rerender } = renderShallow(
                <TerritoriesListManager { ...props } />
            ));
            instance().handleTerritorySearch('Japan');
            component = rerender();
        });

        test('should show filtered list', () => {
            expect(findWithClass(
                component,
                'TerritoriesListManager-territories-list'
            )).toMatchSnapshot();
        });
    });
});
