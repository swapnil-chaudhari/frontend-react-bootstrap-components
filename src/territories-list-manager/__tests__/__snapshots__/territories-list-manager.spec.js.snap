// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TerritoriesListManager> when it renders should contain ListHeader component in TerritoriesListManager-list 1`] = `
<div
  className="TerritoriesListManager-list Opm-thin-grey-border Opm-border-radius"
>
  <ListHeader
    formatMessage={[Function]}
    header="Cleared For Sale"
    messages={Object {}}
    onCountrySearch={[Function]}
    onCountrySearchClose={[Function]}
    onTerritorySelect={[Function]}
    searchQuery=""
    selectedCountriesNumber={0}
    totalCountriesNumber={0}
  />
  <div
    className="TerritoriesListManager-territories-list"
  >
    <TerritoriesList
      continentNameKey="continent"
      expandAll={false}
      formatMessage={[Function]}
      messages={Object {}}
      onCountrySelect={[Function]}
      selectedCountries={Array []}
      territories={Array []}
      territoryIdKey="orchId"
      territoryNameKey="territoryName"
      theOtherListLabel="the other list"
    />
  </div>
</div>
`;

exports[`<TerritoriesListManager> when it renders should contain TerritoriesListManager-move-button  1`] = `
<div
  className="TerritoriesListManager col-xs-6"
  id="foo"
>
  <div
    className="TerritoriesListManager-list Opm-thin-grey-border Opm-border-radius"
  >
    <ListHeader
      formatMessage={[Function]}
      header="Cleared For Sale"
      messages={Object {}}
      onCountrySearch={[Function]}
      onCountrySearchClose={[Function]}
      onTerritorySelect={[Function]}
      searchQuery=""
      selectedCountriesNumber={0}
      totalCountriesNumber={0}
    />
    <div
      className="TerritoriesListManager-territories-list"
    >
      <TerritoriesList
        continentNameKey="continent"
        expandAll={false}
        formatMessage={[Function]}
        messages={Object {}}
        onCountrySelect={[Function]}
        selectedCountries={Array []}
        territories={Array []}
        territoryIdKey="orchId"
        territoryNameKey="territoryName"
        theOtherListLabel="the other list"
      />
    </div>
  </div>
  <div
    className="TerritoriesListManager-undo-message Opm-thin-grey-border hide"
  >
    <UndoLink
      formatMessage={[Function]}
      handleNotificationClose={[Function]}
      handleUndoChanges={[Function]}
      messages={Object {}}
      movedCountriesNumber={0}
    />
  </div>
  <div
    className="TerritoriesListManager-move-button Opm-border-radius hide"
  >
    <MoveCountryButton
      formatMessage={[Function]}
      forward={true}
      messages={Object {}}
      onCountryMove={[Function]}
      selectedCountriesNumber={0}
      theOtherListLabel="the other list"
      theOtherListType="bar"
      thisListType="foo"
    />
  </div>
</div>
`;

exports[`<TerritoriesListManager> when it renders should hide TerritoriesListManager-undo-message when no changes available  1`] = `
<div
  className="TerritoriesListManager-undo-message Opm-thin-grey-border hide"
>
  <UndoLink
    formatMessage={[Function]}
    handleNotificationClose={[Function]}
    handleUndoChanges={[Function]}
    messages={Object {}}
    movedCountriesNumber={0}
  />
</div>
`;

exports[`<TerritoriesListManager> when search is triggered should show filtered list 1`] = `
<div
  className="TerritoriesListManager-territories-list"
>
  <TerritoriesList
    continentNameKey="continent"
    expandAll={true}
    formatMessage={[Function]}
    messages={Object {}}
    onCountrySelect={[Function]}
    selectedCountries={Array []}
    territories={
      Array [
        Object {
          "countryName": "Japan",
        },
      ]
    }
    territoryIdKey="orchId"
    territoryNameKey="countryName"
    theOtherListLabel="the other list"
  />
</div>
`;

exports[`<TerritoriesListManager> when undoNotificationVisible is true kand movedCountriesNumber is greater than 0 should show TerritoriesListManager-undo-message 1`] = `
<div
  className="TerritoriesListManager-undo-message Opm-thin-grey-border"
>
  <UndoLink
    formatMessage={[Function]}
    handleNotificationClose={[Function]}
    handleUndoChanges={[Function]}
    messages={Object {}}
    movedCountriesNumber={2}
  />
</div>
`;
